doc/tools/docgram/orderedGrammar

```diff
@@ -2118,23 +2118,27 @@ rewrite_occs: [
 ]
 
 rewstrategy: [
+| rewstrategy0 ";" rewstrategy0
+| rewstrategy0
+]
```

Comment:
Only one or two `rewstrategy0`s are allowed?  You don't want something like `rewstrategy ";" rewstrategy0` or `LIST1` for the first production?

`rewrite_strat XXX; fail` seems ambiguous: is fail a rewstrategy0 or a tactic?  Even if it's not ambiguous, users are likely to find the overloading of the semicolon confusing.

Reply:
Good remark. This is currently resolved as `rewrite_strat (XXX; fail)`. I don't know how extensively `rewrite_strat` is currently used but I'm afraid it is quite late for a change.

> Only one or two `rewstrategy0`s are allowed? You don't want something like `rewstrategy ";" rewstrategy0` or `LIST1` for the first production?

The entry is left-associative, so it means `rewstrategy0 ";" rewstrategy`. What is the way to tell to `orderedGrammar` which associativity to use?

> `rewrite_strat XXX; fail` seems ambiguous: is fail a rewstrategy0 or a tactic? Even if it's not ambiguous, users are likely to find the overloading of the semicolon confusing.

Good remark. This is currently resolved as `rewrite_strat (XXX; fail)`. I don't know how extensively `rewrite_strat` is currently used but I'm afraid it is quite late for a change.

Reply:
> What is the way to tell to orderedGrammar which associativity to use?

I suggest either a) editing the `rewstrategy` production in common.edit_mlg to `LIST1 rewstrategy1 SEP ";"` on the right hand side or (better) b) defining separate `rewstrategies` and `rewstrategy` productions in the mlg instead of using levels. 
 This will be more readable/meaningful to users.

For now, how about suggesting the workaround `(rewrite_strat XXX); fail` when you want the `fail` tactic?  (in the description of rewrite_strat)  We could fix this syntax by eventually requiring parentheses when there's more than one rewstrategy1 and generating a deprecation warning for a while if the parentheses are missing.

Reply:
I used `rewstrategy` and `rewstrategy_atom` instead.

I added a note about the ambiguity wrt ";" and "fail".

Reply:
Elsewhere in the doc we've often used the plural form to refer to lists of items and the singular form to refer to the underlying item.  Not sure what "atom" means to the user.  But if you really prefer *_atom, it's not worth debating.

Reply:
> Elsewhere in the doc we've often used the plural form to refer to lists of items and the singular form to refer to the underlying item. Not sure what "atom" means to the user. But if you really prefer *_atom, it's not worth debating.

@jfehrle  This is not what we do for [`ltac_expr`](https://coq.github.io/doc/master/refman/proof-engine/ltac.html#grammar-token-ltac_expr), which I believe is the proper comparison here:
![image](https://github.com/coq/coq/assets/396076/5e422c8d-ebb3-41ac-a264-807666133b47)